src/pages/signup/components/SignupForm/index.tsx

```diff
@@ -0,0 +1,123 @@
+import styles from './SignupForm.module.scss';
+import { useRouter } from 'next/router';
+import { useForm } from 'react-hook-form';
+import { zodResolver } from '@hookform/resolvers/zod';
+import { z } from 'zod';
+import axios from '@/apis/axios';
+import { isAxiosError } from 'axios';
+import PageButton from '@/components/Button/PageButton';
+import Input from '@/components/Inputs/Input';
+import setToast from '@/utils/setToast';
+
+type FormValues = {
+  email: string;
+  nickname: string;
+  password: string;
+  passwordConfirm: string;
+};
+
+const schema = z
+  .object({
+    email: z
+      .string()
+      .trim()
+      .min(1, { message: '이메일을 입력해 주세요.' })
+      .email({ message: '이메일 형식으로 작성해 주세요.' }),
+    nickname: z
+      .string()
+      .trim()
+      .min(1, { message: '닉네임을 입력해 주세요.' })
+      .max(10, { message: '열 자 이하로 작성해주세요.' }),
+    password: z
+      .string()
+      .trim()
+      .min(1, { message: '비밀번호를 입력해 주세요.' })
+      .min(8, { message: '8자 이상 입력해주세요.' }),
+    passwordConfirm: z.string().trim().min(1, '비밀번호를 입력해 주세요.'),
+  })
+  .refine((data) => data.password === data.passwordConfirm, {
+    path: ['passwordConfirm'],
+    message: '비밀번호가 일치하지 않습니다.',
+  });
+
+function SignupForm() {
+  const router = useRouter();
+  const {
+    register,
+    formState: { errors, isValid },
+    handleSubmit,
+    setError,
+    reset,
+  } = useForm<FormValues>({
+    resolver: zodResolver(schema),
+    mode: 'all',
+    defaultValues: {
+      email: '',
+      nickname: '',
+      password: '',
+      passwordConfirm: '',
+    },
+  });
+
+  const handleValidSubmit = async (data: FormValues) => {
+    try {
+      await axios.post('/users', data);
+      setToast('success', '가입이 완료되었습니다.');
+      router.push('/login');
+    } catch (e) {
+      if (isAxiosError(e)) {
+        const message =
+          e.response?.data.message || '다른 이메일을 입력해 주세요.';
```

Comment:
```suggestion
        const message =
          e.response?.data.message || '다른 이메일을 입력해 주세요.';
```

`e.response?.data.message`는 어떤 메세지가 들어오는건가요??

Reply:
> `e.response?.data.message`는 어떤 메세지가 들어오는건가요??

API에서 넘겨준 메세지를 출력하는 거예요~

![스크린샷 2024-04-19 103028](https://github.com/19-Takify/19-taskify/assets/98067115/131c6244-2814-494e-80d4-fcbd2f306d34)

Reply:
[여기](https://github.com/19-Takify/19-taskify/pull/32#discussion_r1572425636)와 마찬가지로 서버 에러 메시지는 별도의 파일로 구분해서 관리하는게 좋을거 같은데 어떻게 생각하실까요?!?

Reply:
> [여기](https://github.com/19-Takify/19-taskify/pull/32#discussion_r1572425636)와 마찬가지로 서버 에러 메시지는 별도의 파일로 구분해서 관리하는게 좋을거 같은데 어떻게 생각하실까요?!?

너무 좋은 거 같아요!! 에러 메세지도 중복되는 경우가 많아서 상수로 분리하면 좋겠습니다👍
아래 코드로 분리해 봤는데 멘토님 의견이 궁금합니다!
```ts
export const LENGTH_ERROR_MESSAGE = {
  MAX: {
    TEN: '10자 이하로 입력해 주세요.',
  },
  MIN: {
    EIGHT: '8자 이상 입력해 주세요.',
  },
};

export const AUTH_ERROR_MESSAGE = {
  EMAIL: {
    EMPTY: '이메일을 입력해 주세요.',
    INVALID: '이메일 형식으로 입력해 주세요.',
    DUPLICATE: '이미 사용중인 이메일입니다.',
  },
  PASSWORD: {
    EMPTY: '비밀번호를 입력해 주세요.',
    NOT_MATCH: '비밀번호가 일치하지 않습니다',
  },
  NICKNAME: {
    EMPTY: '닉네임을 입력해 주세요.',
  },
};
```